### PR TITLE
Add com.sun.security.sasl.Provider automatically for Infinispan

### LINKED ISSUE #19195: Infinispan client cannot find `SaslServerFactory` in native mode
Testing the latest Camel Quarkus Infinispan SNAPSHOT with the latest Quarkus Infinispan SNAPSHOT reveals an error in native mode. Seems that Infinispan itests in Quarkus do not use authentication, so maybe this has gone unnoticed since the upgrade to GraalVM 21.2.0. 

The client configuration used in the Camel Infinispan itest is [here](https://github.com/apache/camel-quarkus/blob/quarkus-main/integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java#L60-L66).

I guess there's a typo in the `IllegalStateException` message, 'now' instead of 'not', which makes things kinda confusing.

```
2021-08-03 09:29:57,789 WARN  [io.net.cha.ChannelInitializer] (HotRod-client-async-pool-1-4) Failed to initialize a channel. Closing: [id: 0xa7565909]: java.lang.IllegalStateException: SaslClientFactory implementation now found
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.getSaslClientFactory(ChannelInitializer.java:203)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.initAuthentication(ChannelInitializer.java:162)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.initChannel(ChannelInitializer.java:80)
	at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:129)
	at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:112)
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:938)
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609)
	at io.netty.channel.DefaultChannelPipeline.access$100(DefaultChannelPipeline.java:46)
	at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1463)
	at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1115)
	at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:650)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:514)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:567)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
```

### PR DESCRIPTION
**Pull request by gsmet**
Fixes #19195

### CODE REVIEW

**Comment by zakkak** [comment]
@gsmet can we please get a test for this as well?

@jamesnetherton can you please check to see if this one fixes #19195 for you?

File: extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java [hunk 1]
```diff
@@ -62,6 +63,7 @@
     private static final String META_INF = "META-INF";
     private static final String HOTROD_CLIENT_PROPERTIES = META_INF + File.separator + "hotrod-client.properties";
     private static final String PROTO_EXTENSION = ".proto";
+    private static final String SASL_SECURITY_PROVIDER = "com.sun.security.sasl.Provider";
```
Comment (zakkak) [inline extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java:66]: We use this in https://github.com/quarkusio/quarkus/blob/57d1dae99a635b32951d7e1594f0969fc61dceb1/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java#L118 as well. Would it make sense to gather such definitions in a single place?

**Comment by gsmet** [comment]
I tested it with the quickstart. In the core repo, it's using the non-authenticated route right now. I personally don't have the time to dig how to switch to authenticated there (and if we want to remove all testing for non-authenticated) given the Infinispan servers are started completely differently in the quickstart and in the core repo.

If someone wants to have it tested in the core repo, be my guest :). /cc @karesti 

Note that I don't want us to add a new native IT for that so the only option is to tweak the existing one.

File: extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java:66]: I personally don't think it's worth it.

**Comment by gsmet** [comment]
Oh and I want this merged sooner rather than later given the quickstarts build has been broken for ages and it's not acceptable.

So if we want to add a test, we can do it in a follow-up PR.

**Comment by jamesnetherton** [comment]
> @jamesnetherton can you please check to see if this one fixes #19195 for you?

On PTO for the remainder of the week, but I have this in place on the camel-quarkus side as a [workaround](https://github.com/apache/camel-quarkus/blob/quarkus-main/extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java#L35), so I know it works fine.

**Review by zakkak — APPROVED** [review]

**Comment by zakkak** [comment]
LGTM! Thanks @gsmet

**Pull request merged**